python/pyspark/sql/types.py

```diff
@@ -434,8 +434,8 @@ def toInternal(self, dt: datetime.datetime) -> int:
     def fromInternal(self, ts: int) -> datetime.datetime:
         if ts is not None:
             # using int to avoid precision loss in float
-            return datetime.datetime.utcfromtimestamp(ts // 1000000).replace(
-                microsecond=ts % 1000000
+            return datetime.datetime.fromtimestamp(ts // 1000000, datetime.timezone.utc).replace(
```

Comment:
[datetime.UTC](https://docs.python.org/3/library/datetime.html#datetime.UTC) was added in 3.11, so use `datetime.timezone.utc` to support 3.9/3.10